cl/search/tests/test_import_harvard_pdfs.py

```diff
@@ -0,0 +1,188 @@
+import json
+import os
+import tempfile
+from unittest.mock import MagicMock
+from unittest.mock import mock_open as mock_open_function
+from unittest.mock import patch
+
+from django.core.files.base import ContentFile
+from django.core.management import call_command
+from django.test import override_settings
+
+from cl.search.management.commands.import_harvard_pdfs import Command
+from cl.search.models import Court, Docket, OpinionCluster
+from cl.tests.cases import TransactionTestCase
+
+
+@override_settings(
+    CAP_R2_ENDPOINT_URL="http://test-endpoint",
+    CAP_R2_ACCESS_KEY_ID="test-access-key",
+    CAP_R2_SECRET_ACCESS_KEY="test-secret-key",
+    CAP_R2_BUCKET_NAME="test-bucket",
+    AWS_STORAGE_BUCKET_NAME="test-cl-bucket",
+    AWS_S3_CUSTOM_DOMAIN="test-cl-domain",
+    AWS_DEFAULT_ACL="public-read",
+    AWS_QUERYSTRING_AUTH=False,
+    AWS_S3_MAX_MEMORY_SIZE=16 * 1024 * 1024,
+)
+class TestImportHarvardPDFs(TransactionTestCase):
+    fixtures = ["court_data.json"]
+
+    def setUp(self):
+        super().setUp()
+        from django.conf import settings
+
+        settings.CAP_R2_ENDPOINT_URL = "http://test-endpoint"
+        settings.CAP_R2_ACCESS_KEY_ID = "test-access-key"
+        settings.CAP_R2_SECRET_ACCESS_KEY = "test-secret-key"
+        settings.CAP_R2_BUCKET_NAME = "test-bucket"
+        self.court = Court.objects.get(pk="scotus")
+
+        # Create test dockets and clusters
+        self.dockets = []
+        self.clusters = []
+        for i in range(2):
+            docket = Docket.objects.create(
```

Comment:
Here you could optionally have used DocketFactory and OpinionClusterFactory. Although as it is it is also valid.